pkg/lso/localvolumeset.go

```diff
@@ -0,0 +1,196 @@
+package lso
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	lsoV1alpha1 "github.com/openshift/local-storage-operator/api/v1alpha1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// LocalVolumeSetBuilder provides a struct for localVolumeSet object from the cluster and a localVolumeSet definition.
+type LocalVolumeSetBuilder struct {
+	// localVolumeSet definition, used to create the localVolumeSet object.
+	Definition *lsoV1alpha1.LocalVolumeSet
+	// Created localVolumeDiscovery object.
+	Object *lsoV1alpha1.LocalVolumeSet
+	// Used in functions that define or mutate localVolumeSet definition. errorMsg is processed
+	// before the localVolumeSet object is created
+	errorMsg string
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+}
+
+// NewLocalVolumeSetBuilder creates new instance of LocalVolumeSetBuilder.
+func NewLocalVolumeSetBuilder(apiClient *clients.Settings, name, nsname string) *LocalVolumeSetBuilder {
+	glog.V(100).Infof("Initializing new %s localVolumeSet structure in %s namespace", name, nsname)
+
+	builder := LocalVolumeSetBuilder{
+		apiClient: apiClient,
+		Definition: &lsoV1alpha1.LocalVolumeSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the localVolumeSet is empty")
+
+		builder.errorMsg = "localVolumeSet 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The nsname of the localVolumeSet is empty")
+
+		builder.errorMsg = "localVolumeSet 'nsname' cannot be empty"
+	}
+
+	return &builder
+}
+
+// PullLocalVolumeSet retrieves an existing localVolumeSet object from the cluster.
+func PullLocalVolumeSet(apiClient *clients.Settings, name, nsname string) (*LocalVolumeSetBuilder, error) {
+	glog.V(100).Infof(
+		"Pulling localVolumeSet object name:%s in namespace: %s", name, nsname)
+
+	builder := LocalVolumeSetBuilder{
+		apiClient: apiClient,
+		Definition: &lsoV1alpha1.LocalVolumeSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the localVolumeSet is empty")
+
+		builder.errorMsg = "localVolumeSet 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the localVolumeSet is empty")
+
+		builder.errorMsg = "localVolumeSet 'nsname' cannot be empty"
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("localVolumeSet oject %s doesn't exist in namespace %s", name, nsname)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Get fetches existing localVolumeSet from cluster.
+func (builder *LocalVolumeSetBuilder) Get() error {
```

Comment:
This function has to return 2 values:
func (builder *LocalVolumeSetBuilder) Get() (lsoV1alpha1.LocalVolumeSet, error)